### PR TITLE
serialize replication stats() only when needed

### PR DESCRIPTION
**Pull request by harshavardhana**
## Description
serialize replication stats() only when needed

## Motivation and Context
also, rename '.minio.sys/buckets/replication' to
   '.minio.sys/buckets/.replication'

## How to test this PR?
Just test by running on a fresh path, create
buckets and let the usage scanner run once. 

Wait for 5mins and observe if any data is 
written at `.minio.sys/buckets/replication/<hostname>.stats/` 

If replication is not configured we do not need to
serialize anything.

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Optimization (provides speedup with no functional changes)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [ ] Fixes a regression (If yes, please add `commit-id` or `PR #` here)
- [ ] Unit tests added/updated
- [ ] Internal documentation updated
- [ ] Create a documentation update request [here](https://github.com/minio/docs/issues/new?label=doc-change,title=Doc+Updated+Needed+For+PR+github.com%2fminio%2fminio%2fpull%2fNNNNN)

### CODE REVIEW

File: cmd/bucket-replication-utils.go [hunk 1]
```diff
@@ -638,7 +638,7 @@ type replicationResyncer struct {
 }
 
 const (
-	replicationDir      = "replication"
```
Comment (poornas) [inline cmd/bucket-replication-utils.go:641]: better to make this change as a separate PR - both mrf and stats use this directory. It needs to be backwards compatible to move the stats and mrf files to new path

**Review by poornas — CHANGES_REQUESTED** [review]

File: cmd/bucket-replication-utils.go (hunk 1 above)
Comment (harshavardhana) [reply cmd/bucket-replication-utils.go:641]: No we don't need to do that @poornas - I think these stats can be completely discarded, they will be regenerated soon enough via scanner.

**Review by poornas — APPROVED** [review]

**Comment by minio-trusted** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-erasure.sh | :heavy_check_mark: |
| mint-dist-erasure.sh | :heavy_check_mark: |
| mint-compress-encrypt-dist-erasure.sh | :heavy_check_mark: |
| mint-pools.sh | :heavy_check_mark: |
Deleting image on docker hub
Deleting image locally

**Review by klauspost — APPROVED** [review]

**Comment by klauspost** [comment]
@harshavardhana This will drop stats on upgrade, right? I assume you are ok with that.

**Comment by harshavardhana** [comment]
> @harshavardhana This will drop stats on upgrade, right? I assume you are ok with that.

Yes @klauspost it will be re-populated via scanner in 5mins - something we can live with.
